syntax = "proto3";

package obby.contenttracker;

option go_package = "github.com/obby/content-tracker/proto/generated";

service ContentTracker {
  // Track a single file change
  rpc TrackChange(TrackRequest) returns (TrackResponse);

  // Track multiple files concurrently
  rpc TrackBatch(BatchRequest) returns (stream TrackProgress);

  // Get content hash without storing
  rpc GetContentHash(HashRequest) returns (HashResponse);

  // Generate diff between two files
  rpc GenerateDiff(DiffRequest) returns (DiffResponse);
}

message TrackRequest {
  string file_path = 1;
  string change_type = 2;
  string project_root = 3;
}

message TrackResponse {
  bool success = 1;
  string error = 2;
  string content_hash = 3;
  int64 file_size = 4;
  int64 version_id = 5;
}

message BatchRequest {
  repeated TrackRequest requests = 1;
}

message TrackProgress {
  string file_path = 1;
  bool success = 2;
  string error = 3;
  string content_hash = 4;
  int64 version_id = 5;
}

message HashRequest {
  string file_path = 1;
}

message HashResponse {
  bool success = 1;
  string error = 2;
  string content_hash = 3;
  int64 file_size = 4;
}

message DiffRequest {
  string old_file_path = 1;
  string new_file_path = 2;
  string old_content = 3;
  string new_content = 4;
}

message DiffResponse {
  bool success = 1;
  string error = 2;
  string diff_content = 3;
  int32 lines_added = 4;
  int32 lines_removed = 5;
}

